{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "train = pd.read_csv('77_cancer_proteomes_CPTAC_itraq.csv')\n",
    "clinical = pd.read_csv('clinical_data_breast_cancer.csv')\n",
    "proteins = pd.read_csv('PAM50_proteins.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RefSeq_accession_number</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>TCGA-AO-A12D</th>\n",
       "      <th>TCGA-C8-A131</th>\n",
       "      <th>TCGA-AO-A12B</th>\n",
       "      <th>TCGA-BH-A18Q</th>\n",
       "      <th>TCGA-C8-A130</th>\n",
       "      <th>TCGA-C8-A138</th>\n",
       "      <th>TCGA-E2-A154</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-AO-A12B</th>\n",
       "      <th>TCGA-A2-A0SW</th>\n",
       "      <th>TCGA-AO-A0JL</th>\n",
       "      <th>TCGA-BH-A0BV</th>\n",
       "      <th>TCGA-A2-A0YM</th>\n",
       "      <th>TCGA-BH-A0C7</th>\n",
       "      <th>TCGA-A2-A0SX</th>\n",
       "      <th>263d3f-I.CPTAC</th>\n",
       "      <th>blcdb9-I.CPTAC</th>\n",
       "      <th>c4155b-C.CPTAC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NP_958782</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1</td>\n",
       "      <td>1.096131</td>\n",
       "      <td>2.609943</td>\n",
       "      <td>-0.659828</td>\n",
       "      <td>0.195341</td>\n",
       "      <td>-0.494060</td>\n",
       "      <td>2.765081</td>\n",
       "      <td>0.862659</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.963904</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.065838</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.398560</td>\n",
       "      <td>0.598585</td>\n",
       "      <td>-0.191285</td>\n",
       "      <td>0.566975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NP_958785</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1g</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>2.650422</td>\n",
       "      <td>-0.648742</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.503899</td>\n",
       "      <td>2.779709</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.938210</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.055893</td>\n",
       "      <td>0.658143</td>\n",
       "      <td>-0.547749</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.606697</td>\n",
       "      <td>-0.183918</td>\n",
       "      <td>0.578702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NP_958786</td>\n",
       "      <td>PLEC</td>\n",
       "      <td>plectin isoform 1a</td>\n",
       "      <td>1.111370</td>\n",
       "      <td>2.650422</td>\n",
       "      <td>-0.654285</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.500619</td>\n",
       "      <td>2.779709</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.943919</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.065838</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.186022</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NP_000436</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1c</td>\n",
       "      <td>1.107561</td>\n",
       "      <td>2.646374</td>\n",
       "      <td>-0.632113</td>\n",
       "      <td>0.205377</td>\n",
       "      <td>-0.510459</td>\n",
       "      <td>2.797995</td>\n",
       "      <td>0.866423</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.935355</td>\n",
       "      <td>-0.487772</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.055893</td>\n",
       "      <td>0.655850</td>\n",
       "      <td>-0.552212</td>\n",
       "      <td>-0.392601</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.186022</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NP_958781</td>\n",
       "      <td>NaN</td>\n",
       "      <td>plectin isoform 1e</td>\n",
       "      <td>1.115180</td>\n",
       "      <td>2.646374</td>\n",
       "      <td>-0.640428</td>\n",
       "      <td>0.215413</td>\n",
       "      <td>-0.503899</td>\n",
       "      <td>2.787023</td>\n",
       "      <td>0.870186</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.935355</td>\n",
       "      <td>-0.503853</td>\n",
       "      <td>-0.10668</td>\n",
       "      <td>-0.062523</td>\n",
       "      <td>0.651264</td>\n",
       "      <td>-0.556675</td>\n",
       "      <td>-0.395581</td>\n",
       "      <td>0.603993</td>\n",
       "      <td>-0.167079</td>\n",
       "      <td>0.576747</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 86 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  RefSeq_accession_number gene_symbol           gene_name  TCGA-AO-A12D  \\\n",
       "0               NP_958782        PLEC   plectin isoform 1      1.096131   \n",
       "1               NP_958785         NaN  plectin isoform 1g      1.111370   \n",
       "2               NP_958786        PLEC  plectin isoform 1a      1.111370   \n",
       "3               NP_000436         NaN  plectin isoform 1c      1.107561   \n",
       "4               NP_958781         NaN  plectin isoform 1e      1.115180   \n",
       "\n",
       "   TCGA-C8-A131  TCGA-AO-A12B  TCGA-BH-A18Q  TCGA-C8-A130  TCGA-C8-A138  \\\n",
       "0      2.609943     -0.659828      0.195341     -0.494060      2.765081   \n",
       "1      2.650422     -0.648742      0.215413     -0.503899      2.779709   \n",
       "2      2.650422     -0.654285      0.215413     -0.500619      2.779709   \n",
       "3      2.646374     -0.632113      0.205377     -0.510459      2.797995   \n",
       "4      2.646374     -0.640428      0.215413     -0.503899      2.787023   \n",
       "\n",
       "   TCGA-E2-A154       ...        TCGA-AO-A12B  TCGA-A2-A0SW  TCGA-AO-A0JL  \\\n",
       "0      0.862659       ...           -0.963904     -0.487772      -0.10668   \n",
       "1      0.870186       ...           -0.938210     -0.487772      -0.10668   \n",
       "2      0.870186       ...           -0.943919     -0.487772      -0.10668   \n",
       "3      0.866423       ...           -0.935355     -0.487772      -0.10668   \n",
       "4      0.870186       ...           -0.935355     -0.503853      -0.10668   \n",
       "\n",
       "   TCGA-BH-A0BV  TCGA-A2-A0YM  TCGA-BH-A0C7  TCGA-A2-A0SX  263d3f-I.CPTAC  \\\n",
       "0     -0.065838      0.655850     -0.552212     -0.398560        0.598585   \n",
       "1     -0.055893      0.658143     -0.547749     -0.392601        0.606697   \n",
       "2     -0.065838      0.655850     -0.552212     -0.392601        0.603993   \n",
       "3     -0.055893      0.655850     -0.552212     -0.392601        0.603993   \n",
       "4     -0.062523      0.651264     -0.556675     -0.395581        0.603993   \n",
       "\n",
       "   blcdb9-I.CPTAC  c4155b-C.CPTAC  \n",
       "0       -0.191285        0.566975  \n",
       "1       -0.183918        0.578702  \n",
       "2       -0.186022        0.576747  \n",
       "3       -0.186022        0.576747  \n",
       "4       -0.167079        0.576747  \n",
       "\n",
       "[5 rows x 86 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Complete TCGA ID</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age at Initial Pathologic Diagnosis</th>\n",
       "      <th>ER Status</th>\n",
       "      <th>PR Status</th>\n",
       "      <th>HER2 Final Status</th>\n",
       "      <th>Tumor</th>\n",
       "      <th>Tumor--T1 Coded</th>\n",
       "      <th>Node</th>\n",
       "      <th>Node-Coded</th>\n",
       "      <th>...</th>\n",
       "      <th>PAM50 mRNA</th>\n",
       "      <th>SigClust Unsupervised mRNA</th>\n",
       "      <th>SigClust Intrinsic mRNA</th>\n",
       "      <th>miRNA Clusters</th>\n",
       "      <th>methylation Clusters</th>\n",
       "      <th>RPPA Clusters</th>\n",
       "      <th>CN Clusters</th>\n",
       "      <th>Integrated Clusters (with PAM50)</th>\n",
       "      <th>Integrated Clusters (no exp)</th>\n",
       "      <th>Integrated Clusters (unsup exp)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-A2-A0T2</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>66</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T3</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>0</td>\n",
       "      <td>-13</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-A2-A0CM</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>40</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>-12</td>\n",
       "      <td>-13</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>Basal</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-BH-A18V</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>48</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N1</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>-12</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-BH-A18Q</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>56</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N1</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>-12</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-BH-A0E0</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>38</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T3</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>0</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Complete TCGA ID  Gender  Age at Initial Pathologic Diagnosis ER Status  \\\n",
       "0     TCGA-A2-A0T2  FEMALE                                   66  Negative   \n",
       "1     TCGA-A2-A0CM  FEMALE                                   40  Negative   \n",
       "2     TCGA-BH-A18V  FEMALE                                   48  Negative   \n",
       "3     TCGA-BH-A18Q  FEMALE                                   56  Negative   \n",
       "4     TCGA-BH-A0E0  FEMALE                                   38  Negative   \n",
       "\n",
       "  PR Status HER2 Final Status Tumor Tumor--T1 Coded Node Node-Coded  \\\n",
       "0  Negative          Negative    T3         T_Other   N3   Positive   \n",
       "1  Negative          Negative    T2         T_Other   N0   Negative   \n",
       "2  Negative          Negative    T2         T_Other   N1   Positive   \n",
       "3  Negative          Negative    T2         T_Other   N1   Positive   \n",
       "4  Negative          Negative    T3         T_Other   N3   Positive   \n",
       "\n",
       "                ...                 PAM50 mRNA SigClust Unsupervised mRNA  \\\n",
       "0               ...                 Basal-like                          0   \n",
       "1               ...                 Basal-like                        -12   \n",
       "2               ...                 Basal-like                        -12   \n",
       "3               ...                 Basal-like                        -12   \n",
       "4               ...                 Basal-like                          0   \n",
       "\n",
       "  SigClust Intrinsic mRNA miRNA Clusters methylation Clusters RPPA Clusters  \\\n",
       "0                     -13              3                    5         Basal   \n",
       "1                     -13              4                    4         Basal   \n",
       "2                     -13              5                    5         Basal   \n",
       "3                     -13              5                    5         Basal   \n",
       "4                     -13              5                    5         Basal   \n",
       "\n",
       "   CN Clusters  Integrated Clusters (with PAM50)  \\\n",
       "0            3                                 2   \n",
       "1            4                                 2   \n",
       "2            1                                 2   \n",
       "3            1                                 2   \n",
       "4            1                                 2   \n",
       "\n",
       "   Integrated Clusters (no exp)  Integrated Clusters (unsup exp)  \n",
       "0                             2                                2  \n",
       "1                             1                                1  \n",
       "2                             2                                2  \n",
       "3                             2                                2  \n",
       "4                             2                                2  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Complete TCGA ID</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age at Initial Pathologic Diagnosis</th>\n",
       "      <th>ER Status</th>\n",
       "      <th>PR Status</th>\n",
       "      <th>HER2 Final Status</th>\n",
       "      <th>Tumor</th>\n",
       "      <th>Tumor--T1 Coded</th>\n",
       "      <th>Node</th>\n",
       "      <th>Node-Coded</th>\n",
       "      <th>...</th>\n",
       "      <th>12543</th>\n",
       "      <th>12544</th>\n",
       "      <th>12545</th>\n",
       "      <th>12546</th>\n",
       "      <th>12547</th>\n",
       "      <th>12548</th>\n",
       "      <th>12549</th>\n",
       "      <th>12550</th>\n",
       "      <th>12551</th>\n",
       "      <th>12552</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-A2-A0CM</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>40</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.15361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-BH-A18Q</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>56</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N1</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0481439</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.881872</td>\n",
       "      <td>2.52707</td>\n",
       "      <td>-8.11124</td>\n",
       "      <td>-16.0298</td>\n",
       "      <td>-2.04607</td>\n",
       "      <td>-1.77843</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-3.06975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TCGA-A7-A0CE</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>57</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>0.644347</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.62595</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.30624</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TCGA-D8-A142</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>74</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T3</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>-5.10763</td>\n",
       "      <td>-0.97598</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.50863</td>\n",
       "      <td>-12.3371</td>\n",
       "      <td>-9.54653</td>\n",
       "      <td>-4.06658</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>TCGA-AO-A0J6</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>61</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.04342</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-3.23134</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-3.75362</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 12583 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Complete TCGA ID  Gender  Age at Initial Pathologic Diagnosis ER Status  \\\n",
       "1     TCGA-A2-A0CM  FEMALE                                   40  Negative   \n",
       "3     TCGA-BH-A18Q  FEMALE                                   56  Negative   \n",
       "5     TCGA-A7-A0CE  FEMALE                                   57  Negative   \n",
       "6     TCGA-D8-A142  FEMALE                                   74  Negative   \n",
       "8     TCGA-AO-A0J6  FEMALE                                   61  Negative   \n",
       "\n",
       "  PR Status HER2 Final Status Tumor Tumor--T1 Coded Node Node-Coded   ...     \\\n",
       "1  Negative          Negative    T2         T_Other   N0   Negative   ...      \n",
       "3  Negative          Negative    T2         T_Other   N1   Positive   ...      \n",
       "5  Negative          Negative    T2         T_Other   N0   Negative   ...      \n",
       "6  Negative          Negative    T3         T_Other   N0   Negative   ...      \n",
       "8  Negative          Negative    T2         T_Other   N0   Negative   ...      \n",
       "\n",
       "       12543    12544     12545    12546    12547    12548    12549    12550  \\\n",
       "1        NaN      NaN   1.15361      NaN      NaN      NaN      NaN      NaN   \n",
       "3  0.0481439      NaN -0.881872  2.52707 -8.11124 -16.0298 -2.04607 -1.77843   \n",
       "5   0.644347      NaN   1.62595      NaN      NaN      NaN -1.30624      NaN   \n",
       "6   -5.10763 -0.97598       NaN  2.50863 -12.3371 -9.54653 -4.06658      NaN   \n",
       "8   -1.04342      NaN       NaN      NaN -3.23134      NaN      NaN      NaN   \n",
       "\n",
       "   12551    12552  \n",
       "1    NaN      NaN  \n",
       "3    NaN -3.06975  \n",
       "5    NaN      NaN  \n",
       "6    NaN      NaN  \n",
       "8    NaN -3.75362  \n",
       "\n",
       "[5 rows x 12583 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X = train.drop(['gene_symbol', 'gene_name'], axis=1).transpose()\n",
    "X.index = X.index.map(lambda x: \"TCGA-%s\" % (re.split('[_|-|.]',x)[0]) if bool(re.search(\"TCGA\",x)) is True else x)\n",
    "X = clinical.merge(X, right_index=True, left_on='Complete TCGA ID', how='inner')\n",
    "X.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Complete TCGA ID</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age at Initial Pathologic Diagnosis</th>\n",
       "      <th>ER Status</th>\n",
       "      <th>PR Status</th>\n",
       "      <th>HER2 Final Status</th>\n",
       "      <th>Tumor</th>\n",
       "      <th>Tumor--T1 Coded</th>\n",
       "      <th>Node</th>\n",
       "      <th>Node-Coded</th>\n",
       "      <th>...</th>\n",
       "      <th>PAM50 mRNA</th>\n",
       "      <th>SigClust Unsupervised mRNA</th>\n",
       "      <th>SigClust Intrinsic mRNA</th>\n",
       "      <th>miRNA Clusters</th>\n",
       "      <th>methylation Clusters</th>\n",
       "      <th>RPPA Clusters</th>\n",
       "      <th>CN Clusters</th>\n",
       "      <th>Integrated Clusters (with PAM50)</th>\n",
       "      <th>Integrated Clusters (no exp)</th>\n",
       "      <th>Integrated Clusters (unsup exp)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-A2-A0T2</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>66</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T3</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>0</td>\n",
       "      <td>-13</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-A2-A0CM</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>40</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N0</td>\n",
       "      <td>Negative</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>-12</td>\n",
       "      <td>-13</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>Basal</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-BH-A18V</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>48</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N1</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>-12</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-BH-A18Q</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>56</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T2</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N1</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>-12</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-BH-A0E0</td>\n",
       "      <td>FEMALE</td>\n",
       "      <td>38</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>T3</td>\n",
       "      <td>T_Other</td>\n",
       "      <td>N3</td>\n",
       "      <td>Positive</td>\n",
       "      <td>...</td>\n",
       "      <td>Basal-like</td>\n",
       "      <td>0</td>\n",
       "      <td>-13</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>Basal</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Complete TCGA ID  Gender  Age at Initial Pathologic Diagnosis ER Status  \\\n",
       "0     TCGA-A2-A0T2  FEMALE                                   66  Negative   \n",
       "1     TCGA-A2-A0CM  FEMALE                                   40  Negative   \n",
       "2     TCGA-BH-A18V  FEMALE                                   48  Negative   \n",
       "3     TCGA-BH-A18Q  FEMALE                                   56  Negative   \n",
       "4     TCGA-BH-A0E0  FEMALE                                   38  Negative   \n",
       "\n",
       "  PR Status HER2 Final Status Tumor Tumor--T1 Coded Node Node-Coded  \\\n",
       "0  Negative          Negative    T3         T_Other   N3   Positive   \n",
       "1  Negative          Negative    T2         T_Other   N0   Negative   \n",
       "2  Negative          Negative    T2         T_Other   N1   Positive   \n",
       "3  Negative          Negative    T2         T_Other   N1   Positive   \n",
       "4  Negative          Negative    T3         T_Other   N3   Positive   \n",
       "\n",
       "                ...                 PAM50 mRNA SigClust Unsupervised mRNA  \\\n",
       "0               ...                 Basal-like                          0   \n",
       "1               ...                 Basal-like                        -12   \n",
       "2               ...                 Basal-like                        -12   \n",
       "3               ...                 Basal-like                        -12   \n",
       "4               ...                 Basal-like                          0   \n",
       "\n",
       "  SigClust Intrinsic mRNA miRNA Clusters methylation Clusters RPPA Clusters  \\\n",
       "0                     -13              3                    5         Basal   \n",
       "1                     -13              4                    4         Basal   \n",
       "2                     -13              5                    5         Basal   \n",
       "3                     -13              5                    5         Basal   \n",
       "4                     -13              5                    5         Basal   \n",
       "\n",
       "   CN Clusters  Integrated Clusters (with PAM50)  \\\n",
       "0            3                                 2   \n",
       "1            4                                 2   \n",
       "2            1                                 2   \n",
       "3            1                                 2   \n",
       "4            1                                 2   \n",
       "\n",
       "   Integrated Clusters (no exp)  Integrated Clusters (unsup exp)  \n",
       "0                             2                                2  \n",
       "1                             1                                1  \n",
       "2                             2                                2  \n",
       "3                             2                                2  \n",
       "4                             2                                2  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def evaluate_cluster(X, y, model, ks)\n",
    "    \n",
    "    model.fit(X)\n",
    "        ## The higher (up to 1) the better\n",
    "        print(\"Silhouette Coefficient for k == %s: %s\" % (\n",
    "        k, round(metrics.silhouette_score(data, clusterer.labels_), 4)))\n",
    "        ## The higher (up to 1) the better\n",
    "        print(\"Homogeneity score for k == %s: %s\" % (\n",
    "        k, round(metrics.homogeneity_score(processed['PAM50 mRNA'], clusterer.labels_),4)))  \n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "import sys\n",
    "sys.path.insert(0,'../..')\n",
    "\n",
    "import re\n",
    "import warnings\n",
    "from itertools import combinations\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import scipy.sparse as sps\n",
    "import pandas as pd\n",
    "\n",
    "import sklearn.feature_extraction\n",
    "import sklearn.preprocessing\n",
    "import sklearn.cross_validation\n",
    "import sklearn.metrics\n",
    "import sklearn.feature_selection\n",
    "\n",
    "import sklearn.ensemble\n",
    "import sklearn.naive_bayes\n",
    "import sklearn.svm\n",
    "import sklearn.tree\n",
    "import sklearn.linear_model\n",
    "\n",
    "\n",
    "from helper import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "a"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.4.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
